File: articles/supply-chain-futures/c/supply_chain_resilience.c
#include <stdio.h>

double resilience_score(double cost, double diversification, double buffer, double visibility, double labor, double climate, double traceability, double circularity, double regulation, double recovery) {
    return 0.10 * cost + 0.16 * diversification + 0.14 * buffer + 0.14 * visibility
        + 0.12 * labor + 0.13 * climate + 0.09 * traceability + 0.06 * circularity
        + 0.04 * regulation + 0.02 * recovery;
}

int main(void) {
    printf("Essential Goods Resilience Model resilience: %.4f\n",
           resilience_score(0.52, 0.76, 0.84, 0.74, 0.72, 0.70, 0.76, 0.54, 0.80, 0.86));
    return 0;
}

File: articles/supply-chain-futures/julia/supply_chain_resilience.jl
# Julia standard-library example for supply chain resilience scoring.

profiles = Dict(
    "Lowest-Cost Global Efficiency" => [0.90, 0.34, 0.24, 0.42, 0.36, 0.30, 0.44, 0.26, 0.38, 0.38],
    "Regional Resilience Shift" => [0.58, 0.78, 0.70, 0.66, 0.64, 0.62, 0.64, 0.48, 0.66, 0.74],
    "Essential Goods Resilience Model" => [0.52, 0.76, 0.84, 0.74, 0.72, 0.70, 0.76, 0.54, 0.80, 0.86]
)

function resilience_score(v)
    # v = cost diversification buffer visibility labor climate traceability circularity regulation recovery
    0.10*v[1] + 0.16*v[2] + 0.14*v[3] + 0.14*v[4] + 0.12*v[5] +
    0.13*v[6] + 0.09*v[7] + 0.06*v[8] + 0.04*v[9] + 0.02*v[10]
end

println("profile,supply_chain_resilience")
for (name, values) in sort(collect(profiles))
    println("$(name),$(round(resilience_score(values), digits=4))")
end
